### ISSUE TITLE
systemd-gpt-auto-generator deletes probed partitions

### ISSUE THREAD

**Issue by msekletar** [systemd/systemd#24431]
### systemd version the issue has been seen with

251

### Used distribution

Fedora

### CPU architectures issue was seen on

x86_64

### Component

systemd-gpt-auto-generator

### Expected behaviour you didn't see

systemd-gpt-auto-generator doesn't delete probed partitions.

### Unexpected behaviour you saw

In 75d7e04eb46 we have introduced ability to relinquish partitions that were potentially probed while dissecting images. Relinquished partitions are later deleted before the underlying device that contains the image is closed (which would trigger asynchronous deletion of these partitions).

We have added call to `dissected_image_relinquish()` also to `gpt-auto-generator` and that causes generator to deleted the discovered partitions. On SELinux enabled systems this also causes following AVC message (deleting partitions requires CAP_SYS_ADMIN),

```
type=AVC msg=audit(08/24/2022 06:27:08.956:927) : avc:  denied  { sys_admin } for  pid=44801 comm=systemd-gpt-aut capability=sys_admin  scontext=system_u:system_r:systemd_gpt_generator_t:s0 tcontext=system_u:system_r:systemd_gpt_generator_t:s0 tclass=capability permissive=0 

```

It seems that current behaviour is not optimal because deleting these partitions seems pointless as they would need to be recreated anyway.

### Steps to reproduce the problem

1.  Run SELinux enabled Fedora and make use of systemd-gpt-auto-generator.

**Comment by msekletar** [comment]
@poettering @DaanDeMeyer just following up on the issue that I've mentioned on the call 2 weeks ago.

**Comment by poettering** [comment]
Fix in → #24530

**Comment by QuintinHill** [comment]
This is causing https://bugzilla.redhat.com/show_bug.cgi?id=2141998 in Fedora 37.  In https://github.com/fedora-selinux/selinux-policy/pull/1468 there is a reluctance to allow sys_admin.  @keszybz can the commit fixing this be backported to 251-stable please.
